### DISCUSSION TITLE
Simple FFT

### DISCUSSION THREAD

**Discussion by MoriXTV** [pschatzmann/ESP32-A2DP#249] [Q&A] [answered]
What would be the simplest way to get a single value for audio amplitude for a defined band (approx. 20-300Hz)?
I have mono downmix enabled so I really just need one value for the entire frequency range (approx. 20-300Hz).
This value will be used for controlling the Brightness of a single color LED strip for lighting effects.
I really need help here.
Thank you.

**Comment by pschatzmann** [comment]
If you already use FFT I would suggest to use the max value of magnitude(int idx) for the values where frequency(int idx) is >20 and <300.

An alternative could be to use a FilteredStream() with a bandpass filter and output it to VolumePrint()

**Reply by MoriXTV** [reply]
Thank you for your answer.
I tried using FFT a few times but never got it to work. Could you give an example of how it would be easiest to implement because many projects I've looked at like the Icon64 are to complicated for me to understand with my limited knowledge of C programming.

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you have a look at the [AudioTools project](https://github.com/pschatzmann/arduino-audio-tools) and that you read the related [documentation in the Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/FFT) of the AudioTools.

You can find [a simple FFT example in the audio kit folder](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/streams-audiokit-fft/streams-audiokit-fft.ino): It is just like writing to any Audio Sink!

Here is an [a2dp example using the AudioTools](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/basic-a2dp-i2s/basic-a2dp-i2s.ino). Just replace the write to i2s with a write to the FFT stream.

However if you plan to update a screen with this information, the screen update logic might need to be in it's separate task, so that the audio output does not get delayed. Here is an [example that uses a separate task](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/tests/codecs/test-codec-codec2/test-codec-codec2.ino)

**Comment by MoriXTV** [comment]
Ok I will look into that. How would the FilteredStream() and VolumePrint() Method work?
I'm really sorry I am just starting to learn about coding but I'm just at the beginning.
